### PR TITLE
Manage token

### PR DESCRIPTION
**Pull request by AngeloCG97**
## Manage token expiration 

#367

### What does this PR do?
- Extended time to expired token basic
- When token expired logout user

### How should this be tested?
- Run make run
- Go to localhost:3000
- Change time to expire token in 30 in the file jw.js in line 20 for fast a prove expiration token

### CODE REVIEW

File: hapi/src/utils/jwt.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const create = ({ account, role, ...args }) => {
     jwtConfig.secret,
     'HS256'
   )
-  jwt.setExpiration(new Date().getTime() + 8 * 60 * 60 * 1000)
+  jwt.setExpiration(new Date().getTime() + 30)
```
Comment (JustinCast) [inline hapi/src/utils/jwt.js:20]: Why 30?

File: webapp/src/routes/Home/Home.js [hunk 2]
```diff
@@ -23,86 +24,106 @@ const Home = () => {
...
 
-  const { refetch: getAllOffers } = useQuery(GET_OFFERS_QUERY, { active: true }, { skip: true })
-  const { refetch: getAllBanks } = useQuery(GET_LOCATIONS_QUERY, {}, { skip: true })
-  const { refetch: getAllSponsors } = useQuery(GET_LOCATIONS_QUERY, {}, { skip: true })
+  const { error: getAllOffersError, refetch: getAllOffers } = useQuery(GET_OFFERS_QUERY, { active: true }, { fetchPolicy: 'cache-and-network' })
+  const { refetch: getAllBanks } = useQuery(GET_LOCATIONS_QUERY, {}, { skip: true }, { fetchPolicy: 'cache-and-network' })
+  const { refetch: getAllSponsors } = useQuery(GET_LOCATIONS_QUERY, {}, { skip: true }, { fetchPolicy: 'cache-and-network' })
+
+  useEffect(() => {
+    if (getAllOffersError) {
```
Comment (JustinCast) [inline webapp/src/routes/Home/Home.js:37]: Unnecessary curly braces here

File: webapp/src/routes/Home/Home.js [hunk 3]
```diff
@@ -23,86 +24,106 @@ const Home = () => {
...
+    if (data) {
+      let dataOffers = data.offer
 
-    if (valueSponsorCat !== 'All') {
-      dataOffers = dataOffers.filter(
-        (offer) =>
-          offer.user.location.info.bussines_type.toLowerCase() === valueSponsorCat.toLowerCase()
-      )
-    }
+      if (valueOfferCat !== 'All') {
```
Comment (JustinCast) [inline webapp/src/routes/Home/Home.js:56]: Unnecessary curly braces here

File: webapp/src/routes/Home/Home.js [hunk 4]
```diff
@@ -23,86 +24,106 @@ const Home = () => {
...
+            offer.offer_type.toLowerCase() === valueOfferCat.toLowerCase()
+        )
+      }
 
-    if (valueTokenPrice !== 'All') {
-      dataOffers = dataOffers.filter(
-        (offer) => offer.cost_in_tokens === parseInt(valueTokenPrice)
-      )
-    }
+      if (valueSponsorCat !== 'All') {
```
Comment (JustinCast) [inline webapp/src/routes/Home/Home.js:63]: Unnecessary curly braces here

File: webapp/src/routes/Home/Home.js [hunk 5]
```diff
@@ -23,86 +24,106 @@ const Home = () => {
...
+        dataOffers = dataOffers.filter(
+          (offer) =>
+            offer.user.location.info.bussines_type.toLowerCase() === valueSponsorCat.toLowerCase()
+        )
+      }
+
+      if (valueTokenPrice !== 'All') {
```
Comment (JustinCast) [inline webapp/src/routes/Home/Home.js:70]: Unnecessary curly braces here

File: webapp/src/routes/Home/Home.js [hunk 6]
```diff
@@ -23,86 +24,106 @@ const Home = () => {
...
+        dataOffers = dataOffers.filter(
+          (offer) => offer.cost_in_tokens === parseInt(valueTokenPrice)
+        )
+      }
+
+      if (searchValue !== "") {
```
Comment (JustinCast) [inline webapp/src/routes/Home/Home.js:76]: Unnecessary curly braces here

File: webapp/src/routes/Home/Home.js [hunk 7]
```diff
@@ -23,86 +24,106 @@ const Home = () => {
...
-    if (valueSponsorCat !== 'All') {
-      dataTemp = dataTemp.filter(bank => bank.info.bussines_type.toLowerCase() === valueSponsorCat.toLowerCase())
-    }
+      dataTemp = dataTemp.filter(bank => bank.type === "SPONSOR")
 
-    if (searchValue !== "")
-      dataTemp = dataTemp.filter(
-        (bank) => bank.info.name.toLowerCase().includes(searchValue.toLowerCase())
-      )
+      if (valueSponsorCat !== 'All') {
```
Comment (JustinCast) [inline webapp/src/routes/Home/Home.js:114]: Unnecessary curly braces here

**Review by JustinCast — CHANGES_REQUESTED** [review]

**Review by JeanVegaD — APPROVED** [review]

**Review by JustinCast — CHANGES_REQUESTED** [review]
image
Take a look at this

**Comment by JustinCast** [comment]
> image
> 
> Take a look at this

This happen when you logout, being in the profile

**Review by JustinCast — APPROVED** [review]

**Pull request merged**